source/common/filesystem/filesystem_impl.cc

```diff
@@ -61,6 +63,36 @@ std::string fileReadToEnd(const std::string& path) {
   return file_string.str();
 }
 
+std::string canonicalPath(const std::string& path) {
+  // TODO(htuch): When we are using C++17, switch to std::filesystem::canonical.
+  char* resolved_path = ::realpath(path.c_str(), nullptr);
+  if (resolved_path == nullptr) {
+    throw EnvoyException(fmt::format("Unable to determine canonical path for {}", path));
```

Comment:
Can you cover this case? Perhaps if you just pass realpath() a bogus path? (I'm not sure what the OSX issue was).